app/src/org/commcare/CommCareApplication.java

```diff
@@ -712,6 +712,7 @@ public void onServiceConnected(ComponentName className, IBinder service) {
                         }
                     }
 
+                    CommCareHeartbeatManager.instance().startHeartbeatCommunications();

```

Comment:
I think the HeartbeatManager's lifecycle can be managed entirely within the CommCareService, which would remove the need for the singleton implementation. The CommCareService already manages its lifecycle cleanly, so if it handled these methods inside of the service ("one Heartbeat per session service") it would be equivalent